kafka/pykafka/consumer.py

```diff
@@ -29,15 +133,63 @@ def topic(self):
     def partitions(self):
         return self._partitions
 
+    @property
+    def fetch_message_max_bytes(self):
+        return self._fetch_message_max_bytes
+
     def __iter__(self):
         while True:
-            yield self.consume()
+            message = self.consume(timeout=self._socket_timeout_ms)
+            if not message and self._consumer_timed_out():
+                raise StopIteration
+            yield message
+
+    def _consumer_timed_out(self):
+        if self._consumer_timeout_ms == -1:
+            return False
+        disp = (time.time() - self._last_message_time) * 1000.0
+        return disp > self._consumer_timeout_ms
 
     def consume(self, timeout=None):
         """Get one message from the consumer.
 
         :param timeout: Seconds to wait before returning None
         """
+        owned_partition = self.partition_cycle.next()
+        message = owned_partition.consume(timeout=timeout)
+
+        if self._auto_commit_enable:
+            self._auto_commit()
+
+        if message:
+            self._last_message_time = time.time()
+            return message
+
+    def _auto_commit(self):
+        if not self._auto_commit_enable or self._auto_commit_interval_ms == 0:
+            return
+
+        if (time.time() - self._last_auto_commit) * 1000.0 >= self._auto_commit_interval_ms:
+            self.commit_offsets()
+
+    def commit_offsets(self):
+        """Use the Offset Commit/Fetch API to commit offsets for this
+            consumer's topic
+        """
+        if not self.consumer_group:
+            raise Exception("consumer group must be specified to commit offsets")
+
+        self._last_auto_commit = time.time()
```

Comment:
Should be in the `_auto_commit` function so that a user calling this doesn't interfere with that functionality.